R/pls.R

```diff
@@ -344,7 +344,7 @@ prep.step_pls <- function(x, training, info = NULL, ...) {
     res <- try(rlang::eval_tidy(cl), silent = TRUE)
     if (inherits(res, "try-error")) {
       cli::cli_warn(
-        "{.fn step_pls failed with error: {as.character(res))}.",
```

Comment:
TIL about `cli::format_error()`. This might be too much but I used this in baguette: 

```
msg <- gsub("(\\{)", "\\1\\1", msg)
msg <- gsub("(\\})", "\\1\\1", msg)
msg <- cli::format_error(msg)
cli::cli_abort(c("All of the models failed. Example:", "x" = "{msg}"))
```

Reply:
I think it will be worth it to have a helper function here. Using the above code for helping. There are many places in {recipes} where we `try(rlang::eval_tidy(cl), silent = TRUE)` and it smells a little like we need a helper function

Reply:
https://github.com/tidymodels/recipes/issues/1386